tools/scalajsenv.js

```diff
@@ -864,14 +881,21 @@ initArray(
   const componentBase = componentData.arrayBase || componentData;
   const arrayDepth = componentData.arrayDepth + 1;
 
+  const typeTag = (((1 << 8) | arrayDepth) << 23) | componentBase.typeTag;
+
+  ArrayClass.prototype.$typeTag = typeTag;
+  ScalaJS.ClassData[typeTag] = this;
```

Comment:
OK. Let's keep the existing version, then. In the future we can refine it on the basis of benchmarks.